config/jobs/ppc64le-cloud/builds/kubernetes-postsubmit.yaml

```diff
@@ -35,9 +35,18 @@ postsubmits:
                 rm -rf /usr/local/go
                 tar -C /usr/local -xzf /tmp/golang.tar.gz
                 export PATH=/usr/local/go/bin:$PATH
+                # Storing go version of environment
+                go_version_env=`go version | cut -d ' ' -f4`
```

Comment:
this is what I'm getting on a Mac for a release golang, am I missing something here?

Reply:
@mkumatag The above cut f4 position is applicable only for `devel` golang version.
Here, the go version is **for sure** going to be a devel version as we are getting the go tar from the s3 bucket that we built and pushed.
The comparison at https://github.com/ppc64le-cloud/test-infra/pull/391/files#diff-ee96dfd0e6c95ddf45ff02a9657dd181b529f9f9d187d8bb99bbc8b68f66b8ecR47 would be true only if the binary's golang version matches with the above one.

Reply:
` go env GOVERSION` is giving a complete output when it is master go:
```
# go env GOVERSION
devel go1.21-39c5070712 Thu Jul 6 23:23:41 2023 +0000
```
I shall try to explain why IMO, the cut should be fine.

This change involves two variables:
`go_version_env`: Version of go on the `test-pod` which is for sure going to be devel version and hence f4 should fetch go version value.
`go_version_bin`: Version of binary's go. Only if it is devel version, we would get it in the f3 place. Hence using f3 in the cut command.
If the binary is mistakenly built using the release version, f3 would be null, and therefore condition would fail.

Else, How about grep with a pattern like below?
`go_version_env = go version | grep -Eo 'go[0-9]+\.[0-9]+(\.[0-9]+)?(-[a-zA-Z0-9]+)?([[a-zA-Z0-9]+)?'`
`go_version_bin = go version _output/bin/kubelet | grep -Eo 'go[0-9]+\.[0-9]+(\.[0-9]+)?(-[a-zA-Z0-9]+)?([[a-zA-Z0-9]+)?'`